Controllers/ProjectControllers/Admin/UserController.php

```diff
@@ -0,0 +1,204 @@
+ */
+class UserController extends BaseController
+{
+    private UserModel $userModel;
+
+    private UserView $userView;
+
+    private Validator $dataValidator;
+
+    private ResultMessageModel $msgModel;
+
+    /**
+     * Object for access to $_POST data
+     */
+    private IDataManagement $postInfo;
+
+    /**
+     * Object for access to $_FILE data
+     */
+    private IDataManagement $fileInfo;
+
+    public function __construct()
+    {
+        $this->userModel = new UserModel();
+        $this->userView = new UserView();
+        $this->postInfo = DataRegistry::getInstance()->get('post');
+
+        if ($this->postInfo->isPost()) {
+            $this->dataValidator = new Validator();
+            $this->msgModel = new ResultMessageModel();
+            $this->fileInfo = DataRegistry::getInstance()->get('file');
+        }
+    }
+
+    /**
+     * Render form for registration
+     *
+     * @param array|null $params
+     * @return void
+     */
+    public function registrationAction(array $params = null): void
+    {
+        if (!$this->userModel->isSigned()) {
+            $options = $this->userView->getOptions('Регистрация', 'admin/registration.phtml');
+            $this->userView->render($options);
+        } else {
+            $this->homeLocationByCustomerType();
+        }
+    }
+
+    /**
+     * Create new user
+     *
+     * @param array|null $params
+     * @return void
+     */
+    public function newAction(array $params = null): void
+    {
+        if (!$this->userModel->isSigned()) {
+            if ($this->postInfo->isPost()) {
+                $emptyResult = $this->dataValidator->emptyCheck($this->postInfo->getPostData(), 'registration');
+                if (!in_array(false, $emptyResult)) {
+                    $correctResult = $this->dataValidator->correctCheck($this->postInfo->getPostData());
+                    if (!in_array('', $correctResult)) {
+                        if ($this->fileInfo->isImageSent()) {
+                            if ($this->fileInfo->isImageType()) {
+                                if ($this->fileInfo->isSizeCorrect($params[0])) {
+                                    if (!$this->userModel->newUser($params[0], $correctResult)) {
+                                        $this->adminLocation('/user/registration');
+                                    } else {
+                                        $this->adminHomeLocation();
+                                    }
+                                } else {
+                                    $this->msgModel->setMsg($this->msgModel->notCorrectData('image_size'));
+                                    $this->adminLocation('/user/registration');
+                                }
+                            } else {
+                                $this->msgModel->setMsg($this->msgModel->notCorrectData('image_type'));
+                                $this->adminLocation('/user/registration');
```

Comment:
I can see you use those two lines at least twice. To avoid a code duplication can you reworked it somehow? For example wrap into a private method and call it when you need to.